src/app/qgsattributetabledialog.h

```diff
@@ -69,44 +69,77 @@ class APP_EXPORT QgsAttributeTableDialog : public QDialog, private Ui::QgsAttrib
      */
     void editingToggled();
 
+    /**
+     * @brief Called when the loading of features has been completed
+     */
+    void loadFinished();
+
+    /**
+     * @brief Called while features are being loaded, cancel to abort
+     * @param i
```

Comment:
Yes, sure (that code is not mine: it was already there from the old progress dialog).